src/unix/linux-core.c

```diff
@@ -1045,10 +1268,36 @@ static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
 
 
 uint64_t uv_get_constrained_memory(void) {
-  /*
-   * This might return 0 if there was a problem getting the memory limit from
-   * cgroups. This is OK because a return value of 0 signifies that the memory
-   * limit is unknown.
-   */
-  return uv__read_cgroups_uint64("memory", "memory.limit_in_bytes");
+  uv__cgroups_subsystem_info_t info;
+  uint64_t rc;
+  /* For v2 only. */
+  uint64_t max;
```

Comment:
maybe declare them in the if block?

Reply:
@cjihrig recommended hoisting var declarations to the top of a function from within a while loop, does that also apply to if blocks?

Reply:
either is acceptable, but top of the function seems to be usually preferred in the style guide